src/odemis/gui/cont/acquisition.py

```diff
@@ -2272,50 +2274,126 @@ def _on_ac_done(self, future):
+
+        self.gauge_progress.Hide()  # hide progress bar
+        self._panel.Parent.Layout()
+        self.btn_align.Bind(wx.EVT_BUTTON, self.on_align)
+
+        # FIXME: How to differentiate between simulator and real HW?
+        #   The following two calibrations can also be run with the simulator.
+        #   However, the rest not. How make compartible with both cases?
+        # CALIBRATIONS_SIM and CALIBRATIONS as global var somehow?
+        self.calibrations = [OPTICAL_AUTOFOCUS, IMAGE_TRANSLATION_PREALIGN]
+
+        # check calibration state of system
+        # If backend was not restarted, but only GUI, then the system is in principle still calibrated.
+        # However, so far when closing the GUI the system is also not calibrated anymore.
+        # self.is_aligned.subscribe(self._on_align_state, init=True) # TODO needed?
 
-        self._panel.align_gauge_progress.Hide()
-        panel.btn_align.Bind(wx.EVT_BUTTON, self._on_btn_align)
-        tab_data.main.is_aligned.subscribe(self._on_align_state, init=True)
+        if not self.is_aligned.value:
+            self._reset_calibration_gui()  # display estimated calibration time
 
-    def _on_btn_align(self, evt):
-        # TODO: implement cancellation
+        self._future_connector = None  # connect the future to the progress bar and its label
+
+    def on_align(self, evt):
+        """
+        Start or cancel the calibration when the button is triggered.
+        :param evt: (GenButtonEvent) Button triggered.
+        """
+        # check if cancelled
         if self._tab_data.main.is_acquiring.value:
-            logging.error("Cancelling alignment currently not supported.")
+            logging.debug("Calibration was cancelled.")
+            self.on_cancel(evt)
             return
+
+        # calibrate
         self._tab_data.main.is_acquiring.value = True  # make sure the acquire/tab buttons are disabled
 
-        # Change button to "cancel", show progress bar
-        self._panel.align_gauge_progress.Show()
-        # Label is displayed on the right, but during the acquisition, we want the gauge bar to occupy the
+        self.gauge_progress.Show()  # show progress bar
+        # Label is displayed on the right, but during the calibration, we want the gauge bar to occupy the
         # full available space --> use a panel instead of a spacer which can be hidden (spacers cannot) to
         # make it look nice.
-        self._panel.align_spacer_panel.Hide()
-        self._panel.align_lbl_gauge.Hide()
-        self._panel.btn_align.SetLabel("Cancel")
+        # self._panel.align_lbl_gauge.Hide()  # hide label progress bar TODO put below gauge instead of hide?
+        self._panel.align_spacer_panel.Hide()  # hide space progress bar and progress bar label
+        self._panel.btn_align.SetLabel("Cancel")  # change button label
         self._panel.Layout()
 
+        # TODO loop over calibrations here and then use batch future?
```

Comment:
@pieleric can you comment on this please?

Reply:
Well, `align.fastem.align()` already takes care of wrapping all the calibrations into a single future, so you don't need to do that.

Reply:
Yes indeed, it was just more a thought to keep it the same style as the acquisition, where we have the batch future and the `fastem.acquire()` only takes care of one ROA at a time.